File: CS/WindowsFormsApplication1/Form2.cs
using DevExpress.XtraBars.Docking2010.Views;
using DevExpress.XtraReports.UserDesigner;
using DevExpress.XtraReports.UserDesigner.Native;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form2 : Form
    {
        public Form2()
        {
            InitializeComponent();
            reportDesigner1.AddCommandHandler(new MyClosingReportCommandHandler(reportDesigner1));

        }
    }

    public class MyClosingReportCommandHandler : ICommandHandler
    {
        XRDesignMdiController controller;

        public MyClosingReportCommandHandler(XRDesignMdiController controller)
        {
            this.controller = controller;
        }
        public bool CanHandleCommand(ReportCommand command, ref bool useNextHandler)
        {
            useNextHandler = !(command == ReportCommand.Closing || command == ReportCommand.Close);
            return !useNextHandler;
        }

        public void HandleCommand(ReportCommand command, object[] args)
        {

            for (int i = 0; i < controller.XtraTabbedMdiManager.View.Documents.Count; i++)
            {
                BaseDocument document = controller.XtraTabbedMdiManager.View.Documents[i];

                XRDesignPanelForm designForm = document.Form as XRDesignPanelForm;
                if (designForm != null)
                {
                    XRDesignPanel panel = designForm.DesignPanel;
                    if (panel.ReportState != ReportState.Saved && panel.ReportState != ReportState.None)
                    {
                        MessageBox.Show("Perform report saving here!");
                        panel.ReportState = ReportState.Saved;
                        panel.CloseReport();
                    }
                }

            }
        }
    }

}

File: VB/WindowsFormsApplication1/Form2.vb
Imports DevExpress.XtraBars.Docking2010.Views
Imports DevExpress.XtraReports.UserDesigner
Imports DevExpress.XtraReports.UserDesigner.Native
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace WindowsFormsApplication1
    Partial Public Class Form2
        Inherits Form

        Public Sub New()
            InitializeComponent()
            reportDesigner1.AddCommandHandler(New MyClosingReportCommandHandler(reportDesigner1))

        End Sub
    End Class

    Public Class MyClosingReportCommandHandler
        Implements ICommandHandler

        Private controller As XRDesignMdiController

        Public Sub New(ByVal controller As XRDesignMdiController)
            Me.controller = controller
        End Sub
        Public Function CanHandleCommand(ByVal command As ReportCommand, ByRef useNextHandler As Boolean) As Boolean Implements ICommandHandler.CanHandleCommand
            useNextHandler = Not (command = ReportCommand.Closing Or command = ReportCommand.Close)
            Return Not useNextHandler
        End Function

        Public Sub HandleCommand(ByVal command As ReportCommand, ByVal args() As Object) Implements ICommandHandler.HandleCommand

            For i As Integer = 0 To controller.XtraTabbedMdiManager.View.Documents.Count - 1
                Dim document As BaseDocument = controller.XtraTabbedMdiManager.View.Documents(i)

                Dim designForm As XRDesignPanelForm = TryCast(document.Form, XRDesignPanelForm)
                If designForm IsNot Nothing Then
                    Dim panel As XRDesignPanel = designForm.DesignPanel
                    If panel.ReportState <> ReportState.Saved AndAlso panel.ReportState <> ReportState.None Then
                        MessageBox.Show("Perform report saving here!")
                        panel.ReportState = ReportState.Saved
                        panel.CloseReport()
                    End If
                End If

            Next i
        End Sub
    End Class

End Namespace
